### ISSUE TITLE
Error with rsyslog during install (utopic)

### ISSUE THREAD

**Issue by Snowflake6** [dnschneid/crouton#1391]
I managed to kill my very stable installation by trying to install NFS support... But that's another story.  So I deleted that chroot, and tried to create another, as follows:

chronos@localhost ~/Downloads $ sudo sh ./crouton -t unity -r utopic
WARNING: utopic is an unsupported release.
You will likely run into issues, but things may work with some effort.
Press Ctrl-C to abort; installation will continue in 5 seconds.
Installing utopic-amd64 chroot to /usr/local/chroots/utopic
Downloading latest debootstrap...
2015-01-29 21:31:13 URL:http://anonscm.debian.org/gitweb/?p=d-i/debootstrap.git;a=snapshot;h=HEAD;sf=tgz [61095] -> "-" [1]
Patching debootstrap...
Downloading bootstrap files...
W: Cannot check Release signature; keyring file not available /usr/share/keyrings/ubuntu-archive-keyring.gpg
...

Standard errors regarding utopic being unsupported, okay.  My last chroot was utopic so that shouldn't be an issue.  However, a few minutes later it ends with this:

...
Setting up rsyslog (7.4.4-1ubuntu11.2) ...
The user `syslog' is already a member of `adm'.
runlevel:/var/run/utmp: No such file or directory
invoke-rc.d: initscript rsyslog, action "restart" failed.
dpkg: error processing package rsyslog (--configure):
 subprocess installed post-installation script returned error exit status 1
Setting up cpio (2.11+dfsg-2ubuntu1.1) ...
Setting up libcgmanager0:amd64 (0.32-4ubuntu1.2) ...
Processing triggers for libc-bin (2.19-10ubuntu2.2) ...
Processing triggers for initramfs-tools (0.103ubuntu8) ...
Errors were encountered while processing:
 rsyslog
E: Sub-process /usr/bin/dpkg returned an error code (1)
Failed to complete chroot setup.
Unmounting /mnt/stateful_partition/crouton/chroots/utopic...
chronos@localhost ~/Downloads $

Suggestions?

**Comment by Snowflake6** [comment]
Small update:  thought I may have an out-of-date crouton, so I downloaded the latest and the results didn't change.

**Comment by alexzapatka** [comment]
i had the same issue.  it cropped up when i was updating my chroot (sudo -e ~/Downloads/crouton -u -n utopic).

i think there was a change made to the rsyslog package that makes it broken in the chroot.  looking at the package, i don't really think it is all that required.  so i just removed it:

sudo enter-chroot -n utopic

when prompted to finish upgrading/installing the chroot, i hit "n" to just enter it.  then i did:

sudo apt-get remove rsyslog

it will remove rsyslog and ubuntu-minimal.  then i exited the (by typing exit) and i ran the update command again:

sudo -e ~/Downloads/crouton -u -n utopic

without rsyslog there anymore, the update completes and your chroot will work again.

i figure if they fix this package, or patch crouton to make this package work again in the future i can just reinstall it.

**Comment by Snowflake6** [comment]
Alex, thanks for the confirmation.  Unfortunately, I seem to be stuck.  I eventually removed my old utopic chroot and have started over, and now I can't even get into the partially configured chroot to try and remove the rsyslog package.

I suspect it's because the chroot has never been able to get to the point of even creating users, so it won't let me in.  If I try entering the chroot, I get this:

chronos@localhost ~Downloads $ sudo enter-chroot -n utopic
Entering /mnt/stateful_partition/crouton/chroots/utopic...
A chroot setup script still exists inside the chroot.
The chroot may not be fully set up.
Would you like to finish the setup? [Y/n/d] n
Skipping setup.  You will be prompted again next time.
UID 1000 not found in utopic 
Unmounting /mnt/stateful_partition/crouton/chroots/utopic...
chronos@localhost ~/Downloads $

**Comment by alexzapatka** [comment]
ahhh... yes, because the last thing that is done is the creation of the
user account... so there is no use.  unfortunatly, i think you are stuck
until it is fixed by the maintainers... you could always create a different
release... trusty or something and use that.  i think you can even then
upgrade trusty to to utopic inside the chroot...

make your trusty chroot:

sudo sh -e ~/Downloads/crouton -r trusty -t unity

once it is done, go in to the chroot and go to a terminal and run these:

sudo apt-get update && sudo apt-get dist-upgrade

this gets the latest updates (there shouldn't be any)

then run




On Sun, Feb 1, 2015 at 6:21 PM, Snowflake6 <notifications@github.com> wrote:

> Alex, thanks for the confirmation. Unfortunately, I seem to be stuck. I
> eventually removed my old utopic chroot and have started over, and now I
> can't even get into the partially configured chroot to try and remove the
> rsyslog package.
>
> I suspect it's because the chroot has never been able to get to the point
> of even creating users, so it won't let me in. If I try entering the
> chroot, I get this:
>
> chronos@localhost ~Downloads $ sudo enter-chroot -n utopic
> Entering /mnt/stateful_partition/crouton/chroots/utopic...
> A chroot setup script still exists inside the chroot.
> The chroot may not be fully set up.
> Would you like to finish the setup? [Y/n/d] n
> Skipping setup. You will be prompted again next time.
> UID 1000 not found in utopic
> Unmounting /mnt/stateful_partition/crouton/chroots/utopic...
> chronos@localhost ~/Downloads $
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dnschneid/crouton/issues/1391#issuecomment-72390840>.
>



-- 
 -o)
  /\\    Message void if penguin violated
_\_V    Don't mess with the penguin

**Comment by Snowflake6** [comment]
I hit on another suggestion through a google search for "rsyslog crouton" that brought me back to this very site (but a different issue from 2013).  It suggested starting the chroot with:

sudo enter-chroot -u root

That made me try it again, also with the "-n utopic" flag.  It worked!  I was able to remove rsyslog (and it removed ubuntu-minimal).  Exited the chroot, re-entered without -u root, and said "yes" to the "complete script" question this time.

It's now merrily installing and configuring packages, so that might have fixed things.

I didn't think it was possible to run the dist-upgrade from within a chroot... Or at least, I thought the results might not be as predictable as starting fresh... Hopefully when/if this completes i'll have a working utopic again.

**Comment by Snowflake6** [comment]
Bingo!  sudo startunity now brings up my Unity desktop as before.  Whew!  :)

Thanks for the suggestion that got me on my way!

**Comment by alexzapatka** [comment]
Great!!  And as far as I know, distribution upgrades work in a chroot
fine.  On the main page there is a section about upgrading your distro,
running the crouton script to fix things, then renaming it.  Again, I
haven't tried it myself, but eventually I will... After making a backup.

**Comment by alphaPhi** [comment]
hi, I have the same exact problem.  Could you please repeat step by step the solution?

**Comment by alexzapatka** [comment]
if you are setting up a new chroot, this was the solution offered by the
man himself back in 2013...

Looks like rsyslog is trying to start itself using upstart. A quick fix to
get the thing configured:
1. sudo enter-chroot -u root (decline to run the preparation script)
2. mv /sbin/initctl /sbin/initctl.bak && ln -s /bin/true /sbin/initctl &&
apt-get install -f && mv /sbin/initctl.bak /sbin/initctl && exec
/prepare.sh The
prep script should finish and the chroot will unmount.
3. Use the chroot like normal

you can find the original thread here
https://github.com/dnschneid/crouton/issues/225



On Mon, Feb 2, 2015 at 3:01 PM, Alexandros L Felekidis <
notifications@github.com> wrote:

> hi, I have the same exact problem. Could you please repeat step by step
> the solution?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dnschneid/crouton/issues/1391#issuecomment-72526531>.
>



-- 
 -o)
  /\\    Message void if penguin violated
_\_V    Don't mess with the penguin

**Comment by dnschneid** [comment]
Utopic EOL'd.  Hopefully we can get Vivid fully working before it too EOLs :)
